Example using http does not compile

**V version:** 
```
$ v version
V 0.1.16
$
```

**OS:** 
WSL Ubuntu 18.04 LTS (Native Host Win x64.)

**What did you do?**
Typed this code into `test.v` (originally copied from vlang.io)
```v
import time
import http

fn main() {
        data := http.get('https://vlang.io/utc_now') or {
                println('failed to read data')
                return
        }
        println(data)
        // The original produces: Fn "time.unix" wrong arg #1. Expected "int" (u)  but got "string"
        /*
        t := time.unix(data)
        println(t.clean())
        */
}
```

**What did you expect to see?**
Compile successfully and produce `test` executable binary.
 
**What did you see instead?**
```
$ v test.v
.test.c: In function ‘main’:
.test.c:5576:12: error: ‘string {aka struct string}’ has no member named ‘ok’
  if (!tmp1 .ok) {
            ^
.test.c:5577:20: error: ‘string {aka struct string}’ has no member named ‘error’
  string err = tmp1 . error;
                    ^
.test.c:5584:32: error: ‘string {aka struct string}’ has no member named ‘data’
  string data = *(string*) tmp1 . data;
                                ^
V panic: clang error
```

--- a/examples/links_scraper.v
+++ b/examples/links_scraper.v
@@ -5,7 +5,9 @@
 import http
 
 fn main() {
-	html := http.get('https://news.ycombinator.com')
+	html := http.get('https://news.ycombinator.com') or {
+		println('Failed fetching from URL')
+	}
 	mut pos := 0
 	for {
 		pos = html.index_after('https://', pos + 1)

--- a/examples/links_scraper.v
+++ b/examples/links_scraper.v
@@ -7,6 +7,7 @@ import http
 fn main() {
 	html := http.get('https://news.ycombinator.com') or {
 		println('Failed fetching from URL')
+		return
 	}
 	mut pos := 0
 	for {

--- a/examples/news_fetcher.v
+++ b/examples/news_fetcher.v
@@ -33,7 +33,9 @@ fn (f mut Fetcher) fetch() {
 		id := f.ids[f.cursor]
 		f.cursor++
 		f.mu.unlock()
-		resp := http.get('https://hacker-news.firebaseio.com/v0/item/${id}.json')
+		resp := http.get('https://hacker-news.firebaseio.com/v0/item/${id}.json') or {
+			println('failed to fetch data from /v0/item/${id}.json')
+		}
 		story := json.decode(Story, resp) or {
 			println('failed to decode a story')
 			exit(1)
@@ -48,9 +50,12 @@ fn (f mut Fetcher) fetch() {
 
 // Fetches top HN stories in 8 coroutines
 fn main() {
-	resp := http.get('https://hacker-news.firebaseio.com/v0/topstories.json')
+	resp := http.get('https://hacker-news.firebaseio.com/v0/topstories.json') or {
+		println('failed to fetch data from /v0/topstories.json')
+		return
+	}
 	ids := json.decode( []int, resp) or {
-		println('failed to fetch topstories.json')
+		println('failed to decode topstories.json')
 		return
 	}
 	fetcher := &Fetcher{ids: ids}

--- a/examples/news_fetcher.v
+++ b/examples/news_fetcher.v
@@ -35,6 +35,7 @@ fn (f mut Fetcher) fetch() {
 		f.mu.unlock()
 		resp := http.get('https://hacker-news.firebaseio.com/v0/item/${id}.json') or {
 			println('failed to fetch data from /v0/item/${id}.json')
+			exit(1)
 		}
 		story := json.decode(Story, resp) or {
 			println('failed to decode a story')

--- a/vlib/http/http.v
+++ b/vlib/http/http.v
@@ -28,7 +28,7 @@ pub:
 }
 
 // embed 'http'
-pub fn request(typ, url, data string) ?Response {
+pub fn response(typ, url, data string) ?Response {
 	req := new_request('GET', url, '') or {
 		return error(err)
 	}
@@ -37,14 +37,14 @@ pub fn request(typ, url, data string) ?Response {
 }
 
 pub fn get(url string) ?string {
-	resp := request('GET', url, '') or {
+	resp := response('GET', url, '') or {
 		return error(err)
 	}
 	return resp.body
 }
 
 pub fn post(url, data string) ?string {
-	resp := request('POST', url, data) or {
+	resp := response('POST', url, data) or {
 		return error(err)
 	}
 	return resp.body

--- a/vlib/http/http.v
+++ b/vlib/http/http.v
@@ -28,17 +28,19 @@ pub:
 }
 
 // embed 'http'
-pub fn get(url string) string {
+pub fn get(url string) ?string {
 	if url == '' {
-		println('http: empty get url')
-		return ''
+		return error('http: empty url')
 	}
 	mut req := new_request('GET', url, '')
 	resp := req.do()
 	return resp.body
 }
 
-pub fn post(url, data string) string {
+pub fn post(url, data string) ?string {
+	if url == '' {
+		return error('http: empty url')
+	}
 	req := new_request('POST', url, data)
 	resp := req.do()
 	return resp.body

--- a/vlib/http/http.v
+++ b/vlib/http/http.v
@@ -28,25 +28,32 @@ pub:
 }
 
 // embed 'http'
-pub fn get(url string) ?string {
-	if url == '' {
-		return error('http: empty url')
+pub fn request(typ, url, data string) ?Response {
+	req := new_request('GET', url, '') or {
+		return error(err)
 	}
-	mut req := new_request('GET', url, '')
 	resp := req.do()
+	return resp
+}
+
+pub fn get(url string) ?string {
+	resp := request('GET', url, '') or {
+		return error(err)
+	}
 	return resp.body
 }
 
 pub fn post(url, data string) ?string {
-	if url == '' {
-		return error('http: empty url')
+	resp := request('POST', url, data) or {
+		return error(err)
 	}
-	req := new_request('POST', url, data)
-	resp := req.do()
 	return resp.body
 }
 
-pub fn new_request(typ, _url, _data string) *Request {
+pub fn new_request(typ, _url, _data string) ?Request {
+	if _url == '' {
+		return error('http: empty url')
+	}
 	mut url := _url
 	mut data := _data
 	// req.headers['User-Agent'] = 'V $VERSION'
@@ -56,7 +63,7 @@ pub fn new_request(typ, _url, _data string) *Request {
 		data = ''
 	}
 	// req.headers = new_map(0, sizeof(string))// []string{}
-	return &Request {
+	return Request {
 		typ: typ
 		url: url
 		data: data
@@ -66,12 +73,6 @@ pub fn new_request(typ, _url, _data string) *Request {
 	}
 }
 
-/* 
-fn (req &Request) do() Response {
-	mut resp := Response{}
-	return resp
-}
-*/
 fn (req mut Request) free() {
 	req.headers.free()
 }

--- a/vlib/http/http.v
+++ b/vlib/http/http.v
@@ -28,7 +28,7 @@ pub:
 }
 
 // embed 'http'
-pub fn response(typ, url, data string) ?Response {
+pub fn fetch(typ, url, data string) ?Response {
 	req := new_request('GET', url, '') or {
 		return error(err)
 	}
@@ -37,14 +37,14 @@ pub fn response(typ, url, data string) ?Response {
 }
 
 pub fn get(url string) ?string {
-	resp := response('GET', url, '') or {
+	resp := fetch('GET', url, '') or {
 		return error(err)
 	}
 	return resp.body
 }
 
 pub fn post(url, data string) ?string {
-	resp := response('POST', url, data) or {
+	resp := fetch('POST', url, data) or {
 		return error(err)
 	}
 	return resp.body

